Remote debugging doc 2435 (#433)
harness/developer-hub: docs/continuous-integration/troubleshoot/troubleshooting-ci.md

---
title: Troubleshooting Continuous Integration (CI)

> description: This topic contains general troubleshooting information for error messages and other issues that can arise. 

Reviewer: ```suggestion
description: Troubleshoot common CI issues. 
```

Reviewer: since descriptions get cut off in our docsite cards, you might want to make this shorter.

If you cannot find a resolution, please contact [Harness Support](mailto:support@harness.io) or [Harness Community Forum](https://community.harness.io/).

### Troubleshooting remote builds

> Harness CI includes a Re-run in Debug Mode command for troubleshooting remote builds. This command is available only for the following build types:

Reviewer: ```suggestion
The Harness CI **Re-run in Debug Mode** feature can troubleshoot remote builds. This command is available only for the following build types:
```

* This feature is available only for builds in remote infrastructures: Harness Cloud, Kubernetes, or VMs on remote hosts.

* The build must have failed at a Run step within a CI stage. You must build a pipeline at least once before you can run the build in Debug mode. 
  
  To troubleshoot a pipeline that builds successfully but still needs remote troubleshooting, add a Run step with the CLI command `exit 1` and re-run in Debug mode.

The following steps describe the workflow.
  
1. Navigate to the **Builds**, **Execution**, or **Execution History** page in the Harness UI.

> 2. Choose **Re-run in Debug Mode** in the pull-down menu (**&equiv;**) for the build you want to troubleshoot.

Reviewer: ```suggestion
2. Choose **Re-run in Debug Mode** in the options menu (**&vellip;**) for the build you want to troubleshoot.
```

Reviewer: I believe your unicode char was diff than what is in the UI. The UI shows the vertical ellipsis (⋮).
